sorting/sorting.go

```diff
@@ -0,0 +1,123 @@
+package sorting
+
+import (
+	"bufio"
+	"errors"
+	"log"
+	"os"
+	"regexp"
+	"sort"
+	"strconv"
+	"strings"
+)
+
+// a defined type to simulate a tuple with three elements as sepecified in the associated function
+type Tuple []int
+
+func (tuple *Tuple) appendOne(number int) error {
+	/* This function appends one number to the defined type Tuple if it's not already full (with 3 numbers)
+	otherwise it'll return an error that will be logged using the function HandleErrors */
+	if len(*tuple) > 3 {
+		return errors.New("cannot add anymore numbers")
+	}
+	*tuple = append(*tuple, number)
+	return nil
+}
+
+func ReadingTuplesFromFile(fileName string) []Tuple {
+	/* Reads the input from the given files specified in the argument fileName, checks for errors in the process of
+	opening the file, the reads it line by line, and uses a helper function to cast the strings into ints for the tuple
+	then if the tuple consists of 3 elements (the main condition), it adds it to the list */
+	var listOfTuples []Tuple
+	var file *os.File
+	var err error
+	file, err = os.Open(fileName)
+	handleErrors(err)
+	defer file.Close()
+	var scanner *bufio.Scanner = bufio.NewScanner(file)
+	for scanner.Scan() {
+		var line string = scanner.Text()
+		var tuple Tuple = stringToInt(line)
+		if len(tuple) == 3 {
+			listOfTuples = append(listOfTuples, tuple)
+		}
+	}
+	return listOfTuples
+}
+
+func handleErrors(err error) {
+	// Logging the error passed as an argument to it, checks whether or not everything's ok
+	if err != nil {
+		log.Fatal(err)
```

Comment:
This supports my comment above. If you terminate here, you don't close opened file handles.